### DISCUSSION TITLE
Stopping a Midi Loop

### DISCUSSION THREAD

**Discussion by Lenvdb64** [melanchall/drywetmidi#311] [Q&A] [answered]
Hi all
I am creating a drum pattern in the pattern builder in a step editor.
The notes have been added to a Midifile and I am running this file to test if it is playing back correctly
I have a "Stop" button which I want to click in order to stop the playback, but it seems that while the loop is running I am unable to click this button or any other button on my control form.
The code below for the Playback is on the Click event of the Play button

![image](https://github.com/user-attachments/assets/33073155-9d6c-496b-ac03-fe6af21bcc91)

How did you work around this?

**Comment by melanchall** [comment, marked as answer]
Hi,

First of all, read this article for future issues – [Support](https://melanchall.github.io/drywetmidi/articles/dev/Support.html). From the article you'll learn that code should be given as text, not as a screenshot. It forces me to type your code to show errors instead of just copying/pasting.

Secondly, you definitely need to learn how things you use works. `SpinWait` blocks your main thread which is the UI thread in your case. Why do you need to use `SpinWait`? I suppose you take it from the example from the [Playback – Overview](https://melanchall.github.io/drywetmidi/articles/playback/Overview.html). But it's not a good idea to just copy/paste code without understanding it :-)

So there is no error in the library. There is no issue in using the `Loop` property. But there is an error in your code. You need to define `OutputDevice` and `Playback` as **class fields** (I hope you know C# or VB.NET) and use them across apllication without any `SpinWait`. Don't forget to dispose the device and playback instances.
